dvc/state.py

```diff
@@ -453,20 +453,22 @@ def remove_unused_links(self, used):
             inode = self._from_sqlite(inode)
             path = os.path.join(self.root_dir, relpath)
 
-            if path in used:
-                continue
-
-            if not os.path.exists(path):
+            if path in used or not os.path.exists(path):
                 continue
 
             actual_inode = get_inode(path)
             actual_mtime, _ = get_mtime_and_size(path, self.repo.tree)
 
-            if inode == actual_inode and mtime == actual_mtime:
+            if (inode, mtime) == (actual_inode, actual_mtime):
                 logger.debug("Removing '{}' as unused link.", path)
-                remove(path)
                 unused.append(relpath)
 
+        return unused
+
+    def remove_links(self, unused):
+        for path in unused:
+            remove(os.path.join(self.root_dir, path))
```

Comment:
There was no join in original code, so this might be an error. BTW, paths are rarely rel to dvc root, they are either absolute or relative to cwd.

Reply:
I splitted the function as-is. Original code used `path` which was absolute path. But, we might not need it.